### PR TITLE
Add support for Synapse link connection

### PR DESCRIPTION
**Pull request by houjiaqi21**
Add support for Synapse Link Connection:

- Get-AzSynapseLinkConnection
- Set-AzSynapseLinkConnection
- Start-AzSynapseLinkConnection
- Stop-AzSynapseLinkConnection
- Remove-AzSynapseLinkConnection
- Get-AzSynapseLinkConnectionDetailedStatus
- Set-AzSynapseLinkConnectionLinkTable
- Get-AzSynapseLinkConnectionLinkTable
- Get-AzSynapseLinkConnectionLinkTableStatus
- Update-AzSynapseLinkConnectionLandingZoneCredential

## Checklist

- [X] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [X] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1178
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by wyunchi-ms — COMMENTED** [review]
Please remove nupkg from LocalFeed. We don't accept nupkg in main branch.

**Comment by wyunchi-ms** [comment]
So we will wait for the nupkg published first.

**Comment by wonner** [comment]
> So we will wait for the nupkg published first.

@wyunchi-ms  The SDK has already been published <PackageReference Include="Azure.Analytics.Synapse.Artifacts" Version="1.0.0-preview.15" /> 
@houjiaqi21 please help fix the build error.

File: src/Synapse/Synapse/Az.Synapse.psd1 [hunk 1]
```diff
@@ -228,7 +228,20 @@ CmdletsToExport = 'Get-AzSynapseSparkJob', 'Stop-AzSynapseSparkJob',
...
+               'Get-AzSynapseLinkConnection',
+               'Set-AzSynapseLinkConnection',
+               'Start-AzSynapseLinkConnection',
+               'Stop-AzSynapseLinkConnection',
+               'Remove-AzSynapseLinkConnection',
+               'Get-AzSynapseLinkConnectionDetailedStatus',
+               'Set-AzSynapseLinkConnectionLinkTables',
+               'Get-AzSynapseLinkConnectionLinkTables',
+               'Get-AzSynapseLinkConnectionLinkTablesStatus',
+               'Update-AzSynapseLinkConnectionLandingZoneCredential'
```
Comment (wonner) [inline src/Synapse/Synapse/Az.Synapse.psd1:244]: remove these line

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**